### DISCUSSION TITLE
Is it possible to treat Tilt rotor type with JSBSim?

### DISCUSSION THREAD

**Discussion by GloryENG** [JSBSim-Team/jsbsim#803] [Q&A]
Hello again.

I have got a lot of information through this discussion panel. Thanks in advance.

Nowadays, multicopter types and, furthermore, Vertical-Takeoff-and-landing(VTOL) type vehicles are widely used. 

In this point, is it possible to treat tilt rotor type aircraft with JSBSim? like V-22 Osprey?

I think, all the part including main and tail wing, fuselage are fixed and rigid, and the only part that move during the flight is engine and thruster parts.
In short, Is there a way to change propeller or rotor orientation at run time with JSBSim?
Like landing gear or flap, can we make <traverse> depending on the user input for <orientation> of <thruster>?
or
changing <orient> angle during the flight by user input?

**Comment by gallonmate** [comment]
According to the post linked below in the flight gear forum, runtime engine/propeller angle change is possible and works. Though if you also want helicopter control, you'll have to hack/override various simulation systems to produce "helicopter" controls. Or if you wanted a more simulated approach you can look at the ah1s helicopter example and rotor control system. Although fair warning, the tuning on the a1hs is not perfect. Supposedly a few better helicopter examples exist for flight gear, unfortunately most flight gear helicopters are made with YaSim, a different FDM and not compatible with JSBSim. I think I found 3 different JSBSim flight gear helicopters.

> fdm/jsbsim/propulsion/engine[0]/pitch-angle-rad

> and changing that at run time does affect the propulsion.

https://forum.flightgear.org/viewtopic.php?f=49&t=34973

**Comment by seanmcleod** [comment]
The thruster for an engine has a location and orientation, e.g.

https://github.com/JSBSim-Team/jsbsim/blob/a8c91ec889288a8e5ae127e8eca9bd359e035420/aircraft/737/737.xml#L124-L140

Which can then also be modified at run-time via the relevant properties, i.e. 

https://github.com/JSBSim-Team/jsbsim/blob/a8c91ec889288a8e5ae127e8eca9bd359e035420/src/models/propulsion/FGThruster.cpp#L99-L110
